### ISSUE TITLE
BUG ModuleNotFoundError: No module named 'dm'

### ISSUE THREAD

**Issue by prjemian** [BCDA-APS/apstools#569]
Trying to create an instance of `aps` object:

```python
In [1]: import apstools.devices

In [2]: aps = apstools.devices.ApsMachineParametersDevice(name="aps")
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
~/.conda/envs/bluesky_2022_1/lib/python3.9/site-packages/ophyd/device.py in __get__(self, instance, owner)
    222         try:
--> 223             return instance._signals[self.attr]
    224         except KeyError:

KeyError: 'aps_cycle'

During handling of the above exception, another exception occurred:

ModuleNotFoundError                       Traceback (most recent call last)
```

<details>

```python
xsd6idb% ./blueskyStarter.sh
which: no conda in (/home/beams/USER6IDB:/home/beams/USER6IDB/bin:/usr/local/bin:/bin:/usr/bin:/usr/sbin:/sbin:/usr/share/Modules/bin:/opt/puppetlabs/bin:/APSshare/epics/extensions-base/3.14.12.3-ext1/bin/linux-x86_64:/APSshare/epics/extensions/bin/linux-x86_64:/home/beams/USER6IDB/IPsoftware/bin/linux:/home/beams/S6ADM/bin/bin-linux:/home/beams/S6ADM/spec/bin:/APSshare/spec/pymca/Linux/PyMca4.4.0:/APSshare/spec/cplot/linux-x86/bin:/APSshare/spec/newplot/linux-x86_64/newplot:/opt/X11R6/bin:/home/beams/S6ADM/bin:/home/beams/S4ADM/bin)
dirname: missing operand
Try 'dirname --help' for more information.
CONDA_ACTIVATE = /APSshare/miniconda/x86_64/bin/activate
Python 3.9.7 (default, Sep 16 2021, 13:09:58)
Type 'copyright', 'credits' or 'license' for more information
IPython 7.29.0 -- An enhanced Interactive Python. Type '?' for help.

IPython profile: bluesky
Activating auto-logging. Current session state plus future input saved.
Filename       : /home/beams18/USER6IDB/bluesky/.logs/ipython_console.log
Mode           : rotate
Output logging : True
Raw input log  : False
Timestamping   : True
State          : active
I Wed-15:15:59 - ############################################################ startup
I Wed-15:15:59 - logging started
I Wed-15:15:59 - logging level = 10
I Wed-15:15:59 - /home/beams18/USER6IDB/bluesky/instrument/collection.py
I Wed-15:15:59 - /home/beams18/USER6IDB/bluesky/instrument/mpl/console.py
I Wed-15:16:03 - #### Bluesky Framework ####
I Wed-15:16:03 - /home/beams18/USER6IDB/bluesky/instrument/framework/check_python.py
I Wed-15:16:03 - /home/beams18/USER6IDB/bluesky/instrument/framework/check_bluesky.py
I Wed-15:16:04 - /home/beams18/USER6IDB/bluesky/instrument/framework/initialize.py
I Wed-15:16:07 - using databroker catalog '6idb'
I Wed-15:16:07 - /home/beams18/USER6IDB/bluesky/instrument/framework/metadata.py
I Wed-15:16:07 - #### Devices ####
I Wed-15:16:07 - #### Callbacks ####
I Wed-15:16:07 - /home/beams18/USER6IDB/bluesky/instrument/callbacks/spec_data_file_writer.py
I Wed-15:16:07 - writing to SPEC file: /home/beams18/USER6IDB/bluesky/20211110-151607.dat
I Wed-15:16:07 -    >>>>   Using default SPEC file name   <<<<
I Wed-15:16:07 -    file will be created when bluesky ends its next scan
I Wed-15:16:07 -    to change SPEC file, use command:   newSpecFile('title')
I Wed-15:16:07 - #### Plans ####
I Wed-15:16:07 - #### Utilities ####
I Wed-15:16:07 - /home/beams18/USER6IDB/bluesky/instrument/utils/image_analysis.py
I Wed-15:16:07 - #### Startup is complete. ####

In [1]: import apstools.devices

In [2]: aps = apstools.devices.ApsMachineParametersDevice(name="aps")
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
~/.conda/envs/bluesky_2022_1/lib/python3.9/site-packages/ophyd/device.py in __get__(self, instance, owner)
    222         try:
--> 223             return instance._signals[self.attr]
    224         except KeyError:

KeyError: 'aps_cycle'

During handling of the above exception, another exception occurred:

ModuleNotFoundError                       Traceback (most recent call last)
<ipython-input-2-ccb93dd6ddb6> in <module>
----> 1 aps = apstools.devices.ApsMachineParametersDevice(name="aps")

~/.conda/envs/bluesky_2022_1/lib/python3.9/site-packages/ophyd/device.py in __init__(self, prefix, name, kind, read_attrs, configuration_attrs, parent, **kwargs)
    757         with do_not_wait_for_lazy_connection(self):
    758             # Instantiate non-lazy signals and lazy signals with subscriptions
--> 759             [getattr(self, attr) for attr, cpt in self._sig_attrs.items()
    760              if not cpt.lazy or cpt._subscriptions]
    761

~/.conda/envs/bluesky_2022_1/lib/python3.9/site-packages/ophyd/device.py in <listcomp>(.0)
    757         with do_not_wait_for_lazy_connection(self):
    758             # Instantiate non-lazy signals and lazy signals with subscriptions
--> 759             [getattr(self, attr) for attr, cpt in self._sig_attrs.items()
    760              if not cpt.lazy or cpt._subscriptions]
    761

~/.conda/envs/bluesky_2022_1/lib/python3.9/site-packages/ophyd/device.py in __get__(self, instance, owner)
    223             return instance._signals[self.attr]
    224         except KeyError:
--> 225             return instance._instantiate_component(self.attr)
    226
    227     def __set__(self, instance, owner):

~/.conda/envs/bluesky_2022_1/lib/python3.9/site-packages/ophyd/device.py in _instantiate_component(self, attr)
   1216
   1217         try:
-> 1218             self._signals[attr] = cpt.create_component(self)
   1219             sig = self._signals[attr]
   1220             for event_type, functions in cpt._subscriptions.items():

~/.conda/envs/bluesky_2022_1/lib/python3.9/site-packages/ophyd/device.py in create_component(self, instance)
    176             cpt_inst = self.cls(pv_name, parent=instance, **kwargs)
    177         else:
--> 178             cpt_inst = self.cls(parent=instance, **kwargs)
    179
    180         if self.lazy and hasattr(self.cls, 'wait_for_connection'):

~/.conda/envs/bluesky_2022_1/lib/python3.9/site-packages/ophyd/sim.py in __init__(self, *args, **kwargs)
    171 class SynSignalRO(SynSignal):
    172     def __init__(self, *args, **kwargs):
--> 173         super().__init__(*args, **kwargs)
    174         self._metadata.update(
    175             connected=True,

~/.conda/envs/bluesky_2022_1/lib/python3.9/site-packages/ophyd/sim.py in __init__(self, func, name, exposure_time, precision, parent, labels, kind, **kwargs)
    133         self.exposure_time = exposure_time
    134         self.precision = precision
--> 135         super().__init__(value=self._func(), timestamp=ttime.time(), name=name,
    136                          parent=parent, labels=labels, kind=kind, **kwargs)
    137         self._metadata.update(

~/.conda/envs/bluesky_2022_1/lib/python3.9/site-packages/apstools/_devices/aps_cycle.py in get(self)
     27     def get(self):
     28         if datetime.now().isoformat(sep=" ") >= self._cycle_ends:
---> 29             from ..beamtime.apsbss import api_bss
     30
     31             # only update from data management after the end of the run

~/.conda/envs/bluesky_2022_1/lib/python3.9/site-packages/apstools/beamtime/apsbss.py in <module>
     63
     64 import datetime
---> 65 import dm  # APS data management library
     66 import logging
     67 import os

ModuleNotFoundError: No module named 'dm'

In [3]: !conda list dm
# packages in environment at /home/beams/USER6IDB/.conda/envs/bluesky_2022_1:
#
# Name                    Version                   Build  Channel
adl2pydm                  0+unknown                pypi_0    pypi
aps-dm-api                5.0.3                    py37_1    aps-anl-tag
pydm                      1.11.1             pyhd8ed1ab_0    conda-forge
readme_renderer           24.0             py39h06a4308_0
tqdm                      4.62.3             pyhd3eb1b0_1

In [4]: import dm
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
<ipython-input-4-7715eb2a1ba6> in <module>
----> 1 import dm

ModuleNotFoundError: No module named 'dm'

In [5]:
```

</details>

**Comment by prjemian** [comment]
Somehow, `aps-dm-api` was installed in a python3.7 directory.

```
lib/python3.7/site-packages/aps_dm_api-5.0.3.dist-info:
direct_url.json  INSTALLER  METADATA  RECORD  REQUESTED  top_level.txt  WHEEL

lib/python3.7/site-packages/dm:
aps_bss/             aps_esaf_db/  cat_web_service/  daq_web_service/  __init__.py        __pycache__/
aps_db_web_service/  aps_user_db/  common/           ds_web_service/   proc_web_service/
```

**Comment by prjemian** [comment]
Learned (by private email) that the conda build for *aps-dm-api* using py39 is not yet available (some problem building it).  Need to require `python<3.9` at this time for the *aps-dm-api* package.

**Comment by prjemian** [comment]
Consequence: no need to build CI on py3.9+ until apsbss is migrated out.

**Comment by prjemian** [comment]
@sveseli - Please keep us updated when conda package for *aps-dm-api* becomes available for python 3.9
